File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/FluentXpoUrlOverlay.ts
class FluentXpoUrlOverlay {
    private xpoUrlOverlay: XpoUrlOverlay;

    /*
     * Creates a new instance of the FluentXpoOverlay
     */
    constructor(overlayName: string = "") {
        this.xpoUrlOverlay = new XpoUrlOverlay(overlayName);
    }
    
    /*
     * Returns the wrapped XpoOverlay
     */
    getXpoOverlay(): XpoUrlOverlay { return this.xpoUrlOverlay; }

    /*
     * Sets the index for this object
     */
    setIndex(index: number) {
        this.xpoUrlOverlay.setIndex(index);

        return this;
    }

    /*
     * Sets the location for this overlay
     */
    setLocation(location: UrlGeneratorModule.XpoUrlOverlayLocations) {
        this.xpoUrlOverlay.setOverlayLocation(location);

        return this.xpoUrlOverlay;
    }

    /*
     * Sets the mode for this overlay
     */
    setMode(mode: UrlGeneratorModule.XpoUrlOverlayModes) {
        this.xpoUrlOverlay.setOverlayMode(mode);

        return this.xpoUrlOverlay;
    }

    /*
     * Sets the name for this overlay
     */
    setName(name: string) {
        this.xpoUrlOverlay.setOverlayName(name);

        return this.xpoUrlOverlay;
    }

    /*
     * Sets the operation for this overlay
     */
    setOperation(operation: UrlGeneratorModule.XpoUrlOverlayOperations) {
        this.xpoUrlOverlay.setOverlayOperation(operation);

        return this.xpoUrlOverlay;
    }

    /*
     * Sets the time for this overlay
     */
    setTime(time: UrlGeneratorModule.XpoUrlOverlayTimes) {
        this.xpoUrlOverlay.setOverlayTime(time);

        return this.xpoUrlOverlay;
    }

}

File: sdk/c#/PicarioXPO.RenderAPI/Fluent/FluentXpoUrlOverlay.cs
using PicarioXPO.RenderAPI.FluentInterface;

namespace PicarioXPO.RenderAPI.Fluent
{
    /// <summary>
    /// Used by FluentXPOUrlGenerator for overlay configuration
    /// </summary>
    public sealed class FluentXpoUrlOverlay : IFluentInterface
    {
        /// <summary>
        /// Returns the wrapped XpoObject
        /// </summary>
        public XpoUrlOverlay XpoUrlOverlay
        {
            get
            {
                return xpoUrlOverlay;
            }
        }

        private readonly XpoUrlOverlay xpoUrlOverlay;

        /// <summary>
        /// Creates a new instance of the FluentXpoObject
        /// </summary>
        public FluentXpoUrlOverlay()
        {
            xpoUrlOverlay = new XpoUrlOverlay();
        }

        /// <summary>
        /// Creates a new instance of the FluentXpoObject
        /// </summary>
        public FluentXpoUrlOverlay(string name)
        {
            xpoUrlOverlay = new XpoUrlOverlay { OverlayName = name };
        }

        /// <summary>
        /// Sets the index for this object
        /// </summary>
        public FluentXpoUrlOverlay SetIndex(int index)
        {
            xpoUrlOverlay.Index = index;

            return this;
        }

        /// <summary>
        /// Sets the location for this object
        /// </summary>
        public FluentXpoUrlOverlay SetLocation(XpoUrlOverlayLocation location)
        {
            xpoUrlOverlay.OverlayLocation = location;

            return this;
        }

        /// <summary>
        /// Sets the mode for this object
        /// </summary>
        public FluentXpoUrlOverlay SetMode(XpoUrlOverlayModes mode)
        {
            xpoUrlOverlay.OverlayMode = mode;

            return this;
        }

        /// <summary>
        /// Sets the name for this object
        /// </summary>
        public FluentXpoUrlOverlay SetName(string name)
        {
            xpoUrlOverlay.OverlayName = name;

            return this;
        }

        /// <summary>
        /// Sets the name for this object
        /// </summary>
        public FluentXpoUrlOverlay SetOperation(XpoUrlOverlayOperations operation)
        {
            xpoUrlOverlay.OverlayOperation = operation;

            return this;
        }

        /// <summary>
        /// Sets the time for this object
        /// </summary>
        public FluentXpoUrlOverlay SetTime(XpoUrlOverlayTimes time)
        {
            xpoUrlOverlay.OverlayTime = time;

            return this;
        }
    }
}